### PR TITLE
Board template enhancements

### PR DESCRIPTION
**Pull request by tanya-coding**
- Adds support for environment wide queries;
- Adds support for ad hoc derived columns (including fetching expression automatically);
- Tweaks how operations are handled/translated;
- Tweaks/fixes how filter values are serialized;
- Fixes handling of COUNT(*);
- Adds more variable examples to README;
- Adds support for multi-line query and board descriptions;

### CODE REVIEW

File: hny-client.go [hunk 1]
```diff
@@ -44,7 +44,9 @@ func (c *HoneycombClient) GetBoard(boardId string) (*HoneycombBoard, error) {
 }
 
 func (c *HoneycombClient) GetQuery(dataset string, queryId string) (*HoneycombQuery, error) {
-
+	if dataset == "" {
+		dataset = "__all__"
+	}
```
Comment (tanya-coding) [inline hny-client.go:49]: To support environment wide queries: https://docs.honeycomb.io/api/tag/Queries/

File: main.go [hunk 2]
```diff
@@ -289,16 +344,16 @@ func generateTemplateGoCode(bt *BoardTemplate) (string, error) {
 		tpl += "\t\t\tHideMarkers: " + fmt.Sprint(qt.GraphSettings.HideMarkers) + ",\n"
 		tpl += "\t\t\tPreferOverlaidCharts: " + fmt.Sprint(qt.GraphSettings.OverlaidCharts) + ",\n"
 		tpl += "\t\t},\n"
-		tpl += "\t\tAutoFilter: true,\n"
+		tpl += "\t\tAutoFilter: false,\n"
```
Comment (tanya-coding) [inline main.go:318]: Since we are downloading exact board queries, we shouldn't need to manipulate filters

File: main.go [hunk 3]
```diff
@@ -354,6 +369,33 @@ func firstLetterToUpper(s string) string {
 	return string(r)
 }
 
+func quoteString(str string) string {
+	if strings.Contains(str, "\n") {
+		return "`" + str + "`"
+	}
+	return "\"" + str + "\""
+}
```
Comment (tanya-coding) [inline main.go:388]: Handle multi-line strings (like descriptions)

File: main.go [hunk 4]
```diff
@@ -354,6 +369,33 @@ func firstLetterToUpper(s string) string {
...
+		}
+		str = strings.Join(vals, ",")
+		str = "[]any{" + str + "}"
+	case string:
+		str = quoteString(fmt.Sprint(val))
+	default:
+		str = fmt.Sprint(val)
+	}
+	return str
+}
```
Comment (tanya-coding) [inline main.go:408]: Serialize arrays, strings a numbers based on their type

File: hny-models.go [hunk 5]
```diff
@@ -70,7 +70,7 @@ type QueryHaving struct {
 	CalculateOp string `json:"calculate_op"`
 	Column      string `json:"column"`
 	Op          string `json:"op"`
-	Value       int    `json:"value"`
+	Value       any    `json:"value"`
```
Comment (tanya-coding) [inline hny-models.go:82]: Having value can be any

File: hny-models.go [hunk 6]
```diff
@@ -126,3 +126,38 @@ type VariableSpec struct {
...
+		"FilterOp_NOT_PREFIX":    "does-not-start-with",
+		"FilterOp_HAS_VALUE":     "exists",
+		"FilterOp_NOT_HAS_VALUE": "does-not-exist",
+		"FilterOp_CONTAINS":      "contains",
+		"FilterOp_NOT_CONTAINS":  "does-not-contain",
+		"FilterOp_IN_RESULT":     "in-result",
+		"FilterOp_JOIN_RESULT":   "join-result",
+		"FilterOp_IN":            "in",
+		"FilterOp_NOT_IN":        "not-in",
+	}
```
Comment (tanya-coding) [inline hny-models.go:157]: Add operation mapping

File: README.md [hunk 7]
```diff
@@ -50,7 +50,7 @@ for the variables file.
   "variables": [
     {
       "name": "metrics.cpu.usage",
-      "providers": [
+      "valueProviders": [
```
Comment (tanya-coding) [inline README.md:53]: Fixes example to match what the tool expects

File: README.md [hunk 8]
```diff
@@ -62,15 +62,24 @@ for the variables file.
...
+          "value": "service.name"
+        }
+      ]
+    },
+    {
+      "name": "dc_k8s.pod.uptime.log10",
+      "valueProviders": [
+        {
+          "kind": "AdHocDerivedColumn",
+          "value": "LOG10($k8s.pod.uptime)"
```
Comment (tanya-coding) [inline README.md:82]: Adds example of a formula (aka ad hoc derived column)

File: README.md [hunk 9]
```diff
@@ -62,15 +62,24 @@ for the variables file.
...
     {
-      "name": "metrics.memory.utilization",
-      "providers": [
+      "name": "service.name",
+      "valueProviders": [
         {
-          "kind": "ExactMatch",
-          "value": "metrics.memory.utilization"
+          "kind": "SchemaMapping",
+          "value": "service_name"
```
Comment (tanya-coding) [inline README.md:69]: Adds example of SchemaMapping

**Review by puckpuck — APPROVED** [review]

File: main.go [hunk 10]
```diff
@@ -164,9 +164,23 @@ func loadVariables() ([]VariableSpec, error) {
...
+				if err != nil {
+					fmt.Println(err) // Ignore error and generate template with empty expression
+				} else {
+					fmt.Println("Found DC expression:", dc.Expression)
+					v.ValueProviders[i].Value = dc.Expression
+				}
+			}
+		}
+	}
+	return vars, nil
```
Comment (tanya-coding) [inline main.go:183]: Assumes derived column name matches name of the variable.

**Pull request merged**